compiler/noirc_evaluator/src/ssa/plonky2_gen/mod.rs

```diff
@@ -603,10 +603,75 @@ impl Builder {
                         let (bit_size_b, target_b) = self.get_integer(rhs)?;
                         assert!(bit_size_a == bit_size_b);
 
-                        let div = add_div_mod(&mut self.asm_writer, target_a, target_b).0;
-                        let zero = self.asm_writer.zero();
-                        let target = self.asm_writer.is_equal(div, zero);
-                        Ok(P2Value::make_boolean(target))
+                        let bit_size = usize::try_from(match bit_size_a {
+                            P2Type::Integer(bit_size) => bit_size,
+                            P2Type::Field => FIELD_BIT_SIZE,
+                            _ => {
+                                let message = format!(
+                                    "less than op invoked on arguments of type {:?}",
+                                    bit_size_a
+                                );
+                                return Err(Plonky2GenError::ICE { message });
+                            }
+                        })
+                        .unwrap();
```

Comment:
Could we please have this outlined in a function named `self.get_integer_bitsize(type_of_a)`?